scripts/create-network.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python3
+#
+# kstest-runner - LMC-like program that only does what's necessary for running
+# kickstart-based tests in anaconda
+#
+# Copyright (C) 2011-2015  Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author(s): Radek Vykydal <rvykydal@redhat.com>
+#
+# The script takes one argument - name of libvirt network that will be created.
+# It looks at existing libvirt subnets in 192.168 range and uses the first
+# unassigned /24 subnet (starting from 192.168.100) for the new virtual nated
+# network that would be used by a kickstart test.  The script outputs
+# "<IP address> <netmask> <gateway IP>", the values that can be substituted
+# into kickstart network command for static network configuration.
+
+import libvirt
+import sys
+import xml.etree.ElementTree as ET
+
+if len(sys.argv) != 2:
+    print("Usage: create-network.py <network name>", file=sys.stderr)
+    sys.exit(1)
+
+network_name = sys.argv[1]
+
+conn = libvirt.open()
+
+# Look for free 192.168 subnet assuming 255.255.255.0 netmask.
+# As 'default' network has.
+
+used_subnets = set()
+for network in conn.listAllNetworks():
+    network_xml = network.XMLDesc()
+    root = ET.fromstring(network_xml)
+    for ip in root.iter('ip'):
+        used_subnets.add(ip.attrib['address'][:11])
```

Comment:
Maybe I'm wrong but I see problem here, that this is not atomic operation.
The tests are started on 4 VMs at once so when you read this settings from libvirt it could be before any of the other 3 VM will be started but when you use it a few lines later in the code then 3 of them could be running with their subnet but you are still using the old value.

I think we need some global synchronization mechanism or better to use some global lock.